docker/README-testing.md

```diff
@@ -22,6 +22,10 @@ Consult the [contributing guide][guideComplementSh] for instructions on how to u
 Under some circumstances, you may wish to build the images manually.
 The instructions below will lead you to doing that.
 
+Note that this image can only be built using [BuildKit](https://docs.docker.com/develop/develop-images/build_enhancements/),
+therefore BuildKit needs to be enabled when calling `docker build`. This can be done by
+setting `DOCKER_BUILDKIT=1` in your environment.
+
```

Comment:
I think BuildKit is required to build the base docker image too, not just the testing images? (See #11691.) If so, should we prepend `DOCKER_BUILDKIT=1 ` to the invocations below?

Reply:
> I think BuildKit is required to build the base docker image too, not just the testing images?

As per https://github.com/matrix-org/synapse/issues/13319 it's possible to get to this point without buildkit because you can build the test image based on the one available in docker hub.

> If so, should we prepend `DOCKER_BUILDKIT=1` to the invocations below?

I'm not sure I how this is a consequence of the base image needing buildkit. As mentioned by the PR description, I considered prepending `DOCKER_BUILDKIT=1` to all `docker build` calls but since there are more ways to enable buildkit than setting the environment variable I figured it would be more complete to not do that.

Reply:
Okay, re-reading https://github.com/matrix-org/synapse/issues/13319#issuecomment-1188798558 this makes more sense.

I think I would still complain that `this image` is unclear. There are three images (base, workers, complement) and it's not clear which needs buildkit---all three?

Reply:
Ah right yes, sorry about the typo!